core/src/banking_stage/transaction_scheduler/receive_and_buffer.rs

```diff
@@ -381,7 +381,19 @@ impl TransactionViewReceiveAndBuffer {
     ) -> usize {
         // If not holding packets, just drop them immediately without parsing.
         if matches!(decision, BufferedPacketsDecision::Forward) {
-            return 0;
+            let num_received = packet_batch_message
+                .iter()
+                .map(|packet_batch| {
+                    packet_batch
+                        .iter()
+                        .filter(|packet| !packet.meta().discard())
+                        .count()
+                })
+                .sum();
+            count_metrics.update(|count_metrics| {
+                saturating_add_assign!(count_metrics.num_received, num_received)
+            });
```

Comment:
this makes it more consistent with the other code path; thought they are still not entirely consistent.

As a follow-up to the additional testing of this PR, I plan to refactor this so there's no internal metrics - return a summary of counts and let the caller to aggregation.